### ISSUE TITLE
Gogs should even when database-connection gets lost

### ISSUE THREAD

**Issue by stonerl** [gogits/gogs#2305]
As of this Bug #1761 I came up with the following idea. If for some reason the database becomes unavailable, currently gogs simply does not start. A better solution would be, if gogs is properly configured but only the database isn't available. Gogs should start but show a 500 or 503 Error-Page. It also should try to connect to the configured database every x-seconds/minutes (maybe this could be a config option).

**Comment by bkcsoft** [comment]
Could put Gogs into a maintenance-mode (this would be nice for when running backups as well)

**Comment by noplanman** [comment]
Any update on this?

I have the database on a different server and it's quite annoying to have to SSH into the gogs instance (or alternatively have to make a cronjob) each time the database server needs to get restarted or the connection gets lost.

Would really appreciate the possibility to start even without a DB, if that's possible and still on the menu :wink:

**Comment by credomane** [comment]
I came across this myself when starting/restarting the server. The init scripts in the /script/init/ folder are all missing optional dependencies for mysql and postgresql.

Simply adding the following after the `# Required-Stop` line makes gogs work after reboots. 

    # Should-Start:      mysql postgresql
    # Should-Stop:       mysql postgresql

Just don't forget `insserver gogs` (on debian 8 anyways) after these changes to update boot dependencies.

As for gogs exiting when the DB connection reset/fails instead of retrying is still there.

**Comment by MYDIH** [comment]
Same Here,

I'm running a bunch of VMs and each upgrade/backup/host restart, gogs doesn't start properly because of the SQLVM latency (The SQLVM starts first, way before the gogs one, but still takes too much time and can't be ready before gogs is) ... It's very annoying ...

I see the tag "dont send pull request", is someone actively working on this issue ? It's there since 2015 ...

**Comment by credomane** [comment]
@MYDIH 
I made a pull request that was accepted a while back (https://github.com/gogits/gogs/pull/4561). It doesn't solve the issue in gogs itself. My PR updated the linux gogs init scripts to wait until "mysql postgresql" have started before starting gogs. You might/should be able to add "SQLVM" or whatever your database is to the init scripts in a similar manner.

**Comment by noplanman** [comment]
@credomane That only does something if the database server is running on the same machine, right?

**Comment by credomane** [comment]
>That only does something if the database server is running on the same machine, right?

@noplanman That is correct. My PR just added the changes mentioned in my [Jun 14th](https://github.com/gogits/gogs/issues/2305#issuecomment-308524827) comment and only applies when the DB that GOGS connects to is on the same machine.

BTW, it would seem the comment I replied to from Nov 20th has disappeared completely. My previous comment doesn't make much sense without it. Oh well.

**Comment by MYDIH** [comment]
I deleted my previous comment because I saw this issue was linked in the next milestone ...

Anyway thanks for the information but @noplanman is right, I don't think it's going to help because I do not have an event "SQL server is up" that gogs could wait for. I may be able to mimic one with a script running on systemd or something but this would be quite time consuming for something a bit hacky ...

Thanks again for your comments
